### DISCUSSION TITLE
Supporting 3D volumes in TEKSI?

### DISCUSSION THREAD

**Discussion by sjib** [teksi/Home#104] [Ideas]
> CURVEPOLYGONZ is a 2D polygon made of straights and arcs with XYZ coordinates, but it is not a 3D object.
in order to support a 3D volume we should use
ALTER TABLE tdh_od.structure ADD COLUMN geometry3d_geometry geometry('POLYHEDRALSURFACEZ', :SRID);

In https://github.com/teksi/district_heating/issues/10#issuecomment-2122446778 the question is raised to start modelling 3D volumes in TEKSI.

Currently we are supporting 2.5 D polylignes and polygons.

Should this be changed in the future? 

- [ ] Quickly, within the next years? 
- [ ] When the INTERLIS data models start to support this?
- [ ] ...

What influence would this have in day to day operation and field operation / measuring? Costs? Processes?

What is your opinion about this? Please add your thoughts, needs and worries about this here.

**Comment by ponceta** [comment]
We currently have integrated curve polygon z to manage :

- [ ] curves
- [ ] 3d nodes information

Which led us to many issues and functionnalities that we are still figuring to solve in QGIS. (last one is the add part of a curved object which is yet unsupported by QGIS). We chose Z because snapping a 3 polyline on  a PolygonZ surface enabled gathering the Z information.

I think starting to discuss about real 3D items as views could be a start (shaping these 3D Forms through the actual attributes could bring us further to the possibility of having the use of 3D in the future).

But for now I would suggest to keep to the basics for the users day-to-day needs in terms of postgreSQL geometry types to make the editing as smooth as possible.

The main philosophy was and still is that whenever we come back from the field with 3D GPS data we can store and not lose the 3D information.
